manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala

```diff
@@ -223,29 +246,54 @@ object DbClient {
       sql"""delete from transfer_requests where id = $id""".update.run.void
         .transact(transactor)
 
+    override def updateTransfers(summary: List[(UUID, TransferSummary)]): IO[Unit] = {
+      val newStatuses = summary.map {
+        case (id, s) =>
+          val status = s.result match {
+            case TransferResult.Success          => TransferStatus.Succeeded
+            case TransferResult.TransientFailure => TransferStatus.Retrying
+            case TransferResult.FatalFailure     => TransferStatus.Failed
+          }
+          (status, s.info, id)
+      }
+
+      Update[(TransferStatus, Option[Json], UUID)](
+        "update transfers set status = ?, info = ? where id = ?"
+      ).updateMany(newStatuses)
+        .void
+        .transact(transactor)
+    }
+
+    override def getResubmitInfoForTransfers(
+      transferIds: NonEmptyList[UUID]
+    ): IO[List[DbClient.ResubmitInfo]] = {
+      val query = fr"""select t.id, q.request_topic, t.body from
+            transfers t
+            left join transfer_requests r on t.request_id = r.id
+            left join queues q on r.queue_id = q.id
+            where""" ++ doobie.Fragments.in(fr"t.id", transferIds)
+
+      query.query[(UUID, String, Json)].to[List].transact(transactor)
+    }
+
     /**
       * Build a transaction which will insert all of the rows (top-level and individual)
       * associated with a batch of transfer requests.
       */
     private def insertRequest(
       queueId: UUID,
       requestId: UUID,
-      transferIds: List[UUID]
+      transfersById: List[(UUID, Json)]
     ): ConnectionIO[Unit] = {
       val insertRequest =
-        sql"""insert into transfer_requests (id, queue_id, transfer_count)
-              values($requestId, $queueId, ${transferIds.length})""".update.run.void
+        sql"""insert into transfer_requests (id, queue_id) values ($requestId, $queueId)""".update.run.void
 
-      val insertTransfers = Update[UUID](
-        s"insert into transfers (id, request_id, status) values (?, '$requestId', 'submitted')"
-      ).updateMany(transferIds).void
+      val insertTransfers = Update[(UUID, Json)](
+        s"""insert into transfers (id, request_id, status, body, info)
+            values (?, '$requestId', '${TransferStatus.Submitted.entryName}', ?, NULL)"""
+      ).updateMany(transfersById).void
```

Comment:
Should we sanity check the insert return to make sure it says it inserted the same number we submitted?